File: LibraryBackend/Library.Models/Entities/Complaint.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Library.Models.Entities
{
    public class Complaint
    {
        public int Id { get; set; }
        public int SenderId { get; set; } 
        public User Sender { get; set; }
        public int TargetId { get; set; }  
        public User Target { get; set; }
        public string Reason { get; set; }
        public DateTime ComplaintDate { get; set; }
        public bool IsResolved { get; set; }
    }
}

File: LibraryBackend/UI/elibrary_desktop/lib/models/complaint.dart
import 'package:json_annotation/json_annotation.dart';

import 'user.dart';
part 'complaint.g.dart';

@JsonSerializable()
class Complaint {
  int? id;
  User? sender;
  User? target;
  String? reason;
  DateTime? complaintDate;
  bool? isResolved;

  Complaint(this.id, this.sender, this.target, this.reason, this.complaintDate, this.isResolved);

  factory Complaint.fromJson(Map<String, dynamic> json) => _$ComplaintFromJson(json);
  Map<String, dynamic> toJson() => _$ComplaintToJson(this);
}